io/src/main/scala/sbt/io/Using.scala

```diff
@@ -69,9 +69,9 @@ object Using {
       def close(s: T) = closeF(s)
     }
 
-  def file[T <: AutoCloseable](openF: File => T): OpenFile[T] = file(openF, closeCloseable)
+  def file[T <: AutoCloseable](openF: File => T): Using[File, T] = file(openF, closeCloseable)
```

Comment:
What would be the downstream impact here?
If someone wrote `val x = Using.file(...)` in their plugin, would this change cause runtime issue?

Reply:
If the plugin isn't recompiled we get a linking error when loading the class. I think that's expected tho:
```
java.lang.NoSuchMethodError: sbt.io.Using$.file(Lscala/Function1;Lscala/Function1;)Lsbt/io/OpenFile;
```

As far as source-level compatibility goes, I'm not even sure it's possible to construct code that would break due to this. We return the same object with a looser static type, so this can only lead to type errors. What I could think of is using a mutable reference:
```
var x = Using.file(...)
x = new Using[File, _] { ... }
```
but this works after the change and not before. (because type inference types `x` as `OpenFile` and thus rejects `Using` objects). I have no formal proof, but since OpenFile is private outside code cannot reference it directly, and it has no public methods that Using doesn't have so I'm pretty confident it should be fine honestly.

Reply:
> If the plugin isn't recompiled we get a linking error when loading the class. 

sbt 1 is published with Semantic Versioning, so we intend to support all plugins published against sbt 1.0 to work throughout the 1.x series. We can create a branch for sbt 2.0, so we'll remember to merge it in the future.

Reply:
Got it. If that's the rule, then so be it. Would you be OK with the `@usecase` solution I mentioned above? That would help the initial issue and could be a transitional fix until 2.0 comes around. (which I will target in another PR)